startup.py

```diff
@@ -0,0 +1,340 @@
+        if os.path.isdir(search_path):
+            # Get the list of *.syncfg files in this directory
+            synergy_configs.extend([
+                os.path.join(search_path, f) for f in os.listdir(search_path)
+                if f.startswith("Maya") and f.endswith(".syncfg")
+            ])
+
+        elif os.path.isfile(search_path):
+            file_name = os.path.basename(search_path)
+            if file_name.find("Maya") > -1 and file_name.endswith(".syncfg"):
+                # Add the specified Synergy Config file directly to the list of paths.
+                synergy_configs.append(search_path)
+
+    return synergy_configs
+
+def _synergy_software_versions(logger, versions=None, display_name=None, icon=None):
+    """
+    Creates SoftwareVersion instances based on the Synergy configuration
+    data from Synergy Config (.syncfg) files found in the local environment.
+
+    :param logger: Logger instance to use to log messages.
+    :param versions: (optional) Specific list of version numbers (as strings) to
+                     find matching executables for.
+    :param display_name: (optional) Specify label to use for all
+                         SoftwareVersions found.
+    :param icon: (optional) Specify icon to use for all
+                 SoftwareVersions found.
+
+    :returns: List of :class:`SoftwareVersion` SoftwareVersion instance.
+    """
+    # Get the list of Maya*.syncfg files in the local environment
+    configs = _synergy_config_files()
+    if not configs:
+        logger.debug(
+            "Unable to determine Autodesk Synergy paths for platform "%
+            sys.platform
+        )
+        return []
+
+    logger.debug("Found Autodesk Synergy Maya config files : %s" % configs)
+    # Determine the list of SoftwareVersion to return from the list
+    # of configurations found and the list of versions requested.
+    sw_versions = []
+    for config in configs:
+        try:
+            # Parse the Synergy Config file as XML
+            doc = XML_ET.parse(config)
+        except Exception, e:
+            raise TankError(
+                "Caught exception attempting to parse [%s] as XML.\n%s" % (config, e)
+            )
+
+        try:
+            # Find the <Application> element that contains the data
+            # we want.
+            app_elem = doc.getroot().find("Application")
+            if app_elem is None:
+                logger.warning(
+                    "No <Application> found in Synergy config file '%s'." % config
+                )
+                continue
+
+            # Convert the element's attribute/value pairs to a dictionary
+            synergy_data = dict(app_elem.items())
+        except Exception, e:
+            raise TankError(
+                "Caught unknown exception retrieving <Application> data from %s:\n%s" %
+                (config, e)
+            )
+
+        if versions and synergy_data["NumericVersion"] not in versions:
```

Comment:
i think it would be great if we could report in the debug log at some point exactly which versions we have detected. This could make it easier to diagnose a case when a client has specified a particular version number and for some reason this isn't detected.